project/apps/web/app/search/page.tsx

```diff
@@ -0,0 +1,237 @@
+'use client';
+
+import { MatchCancelDto, MatchRequestMsgDto } from '@repo/dtos/match';
+import { useMutation } from '@tanstack/react-query';
+import { AnimatePresence, motion } from 'framer-motion';
+import { useRouter, useSearchParams } from 'next/navigation';
+import { useEffect, useRef, useState } from 'react';
+
+import { Button } from '@/components/ui/button';
+import { useToast } from '@/hooks/use-toast';
+import { cancelMatch, createMatch } from '@/lib/api/match';
+import useSocketStore from '@/stores/useSocketStore';
+
+const Search = () => {
+  const { toast } = useToast();
+  const router = useRouter();
+  const searchParams = useSearchParams();
```

Comment:
could possibly add some logic to redirect if searchParams is not of expected type or empty